services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/ApplicationConfig.java

```diff
@@ -323,6 +341,26 @@ public ObjectFactoryCreatingFactoryBean httpServerFactory() {
         return factory;
     }
 
+    /**
+     * Creates a new instance of an auth handler to provide basic authentication for the 
+     * Hono's Device Registry Management APIs.
+     * <p>
+     * This creates an instance of the {@link HonoBasicAuthHandler} with an auth provider of type
+     * {@link MongoAuth} to verify the credentials.
+     *
+     * @return The auth handler.
+     * @see <a href="https://vertx.io/docs/vertx-auth-mongo/java/">Mongo auth provider docs</a>
+     */
+    @Bean
+    @Scope("prototype")
+    @ConditionalOnProperty(name = "hono.registry.http.authenticationRequired", havingValue = "true", matchIfMissing = true)
```

Comment:
instead of this conditional, you could simply pass in the config object as a param and check its *authenticationRequired* property to determine, if an `AuthHandler` needs to be created or not ...

Reply:
The `ConditionalOnProperty` has been replaced now with the above option.